WordPress/classic-editor
# Add an endpoint to allow easy retrieval of plugin settings

- Add an endpoint to retrieve some of the plugin settings for the current authenticated user.

- Expose the `classic-editor-remember` post meta to the REST API (`/wp/v2/posts`).

### The Post Meta

The `classic-editor-remember` post meta was already used by the plugin to keep track of the last editor used to edit a post.
This PR explicitly registers it in order to show it in the REST API (`/wp/v2/posts` endpoints).

#### Values

- `block-editor`
- `classic-editor`
- Empty string if the post hasn't been edited yet since the plugin has been installed.

### The Endpoint

`/classic-editor/v1/settings`

Return the current user's selected editor, and if they are allowed to switch to a different one.

#### Response

- `can_switch_editors` (bool): if the user is allowed to switch editors (e.g. from `block` to `classic`).
- `selected_editor` (string): the user's selected editor (possible values: `block`, `classic`).

Example:
```json
{
  "can_switch_editors": true,
  "selected_editor": "classic"
}
```

### Testing Instructions

- Try changing the plugin settings, both in [Settings -> Writing] and in [Users -> Your Profile].
- Call the `/classic-editor/v1/settings` endpoint as an authenticated user, and make sure the response is consistent with the plugin settings.
- With the plugin set to "Allow users to switch editors", edit a post. Note the editor used.
- Call the `/wp/v2/posts/:post_id` endpoint, and make sure the `meta` object contains a `classic-editor-remember` property, and its value is the editor used to open the post.
